bokeh/resources.py

```diff
@@ -290,7 +290,7 @@ def _collect_external_resources(self, resource_attr):
                 external_resources.extend(external)
 
         # Return only unique external resources
-        return list(set(external_resources))
+        return list(external_resources)
```

Comment:
You actually want to deduplicate resources (alphabetically or by model alphabetically). The result should be sorted in some way. You can gain some insight by looking into custom models code below.